### DISCUSSION TITLE
Hooking into Copy (Project) Action

### DISCUSSION THREAD

**Discussion by N1k145** [eclipse-platform/eclipse.platform.ui#2155] [Q&A] [answered]
We are currently looking into hooking into the Copy Project Action, but we are not able to find an API for this.

I digged a bit through PDE and Platform and found LTK Refactoring and what I noticed is that:

- LTK Supports Copy 
- Platform UI is triggering LTK for Rename, Move and Delete
- PDE is using it to hook into Rename and change the Manifest File if the Project is renamed.
- But when you copy a Project in PDE the Manifest File is not updated, which also feels a bit like a gap and is exactly the same problem we are trying to solve.

Does anybody know why LTK is not used for Copying Resources? Or does anybody have a different Idea on how to solve this?

I would try to add the call to LTK to the org.eclipse.ui.ide.undo.CopyProjectOperation, or are there other Ideas?

**Comment by mickaelistria** [comment, marked as answer]
> I would try to add the call to LTK to the org.eclipse.ui.ide.undo.CopyProjectOperation, or are there other Ideas?

This seems like the right thing to try.

**Reply by N1k145** [reply]
@mickaelistria thank you for your input, I tried it an from my perspective it looks quite good now. It would be great if you could have a quick look at #2262 where I implemented a first try.
